src/styles/main.scss

```diff
@@ -0,0 +1,87 @@
+$main-color: #eee;
+$hover-color: #ffbfcb;
+$font-family: Arial, Halvetica, san-serif;
+$week-days: (
+  'mon': 0,
+  'tue': 1,
+  'wed': 2,
+  'thu': 3,
+  'fri': 4,
+  'sat': 5,
+  'sun': 6,
+);
+$columns: 7;
+$calendar-size: 100px;
+$number: 31;
+$days-gap: 1px;
+
+* {
+  box-sizing: border-box;
+}
+
+body {
+  margin: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 100vh;
+}
+
+.calendar {
+  width: ($calendar-size * $columns) + (6 * $days-gap);
+  flex-wrap: wrap;
+  display: flex;
+  gap: $days-gap;
+
+  &__day {
+    background-color: $main-color;
+    border: 1px solid black;
+    justify-content: center;
+    align-items: center;
+    display: flex;
+    width: $calendar-size;
+    height: $calendar-size;
+    position: relative;
+
+    @for $number from 1 through 31 {
+      &:nth-child(#{$number})::before {
+        content: '#{$number}';
+        font-family: $font-family;
+        font-weight: 500;
+        font-size: 30px;
+      }
+    }
+
+    &:hover {
+      animation: move 0.5s ease alternate forwards;
+      background-color: $hover-color;
+      cursor: pointer;
+    }
+  }
+}
+
+@each $day, $i in $week-days {
```

Comment:
The variable names $day and $i in the @each directive are not descriptive enough. Consider using more descriptive names such as $weekDay and $index to improve code readability.